### PR TITLE
Added SanityCheck class

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation

Added sanity check class to verify that passed in params are as expected (non null, non empty)

## Description

Adds a new utility class similar to the one we created for the android SDK  to perform simple sanity checks.

## Progress

- [x] Add SanityCheck class
- [x] Add Unit Tests

### CODE REVIEW

**Comment by wojta** [comment]
@ziccardi check `Contract` class that Is already there for .NET. We should use that instead.

**Comment by ziccardi** [comment]
@wojta thanks! Closing this.

**Comment by ziccardi** [comment]
Reopened this pull request since:

1. Looks like Contract are not supported anymore (https://stackoverflow.com/questions/40767941/does-vs2017-work-with-codecontracts)
2. They require a VS plugin, but I couldn't find it for the MacOS (https://stackoverflow.com/questions/8300815/contract-requires-usage)

File: Core/AeroGear.Mobile/Core/Utils/SanityCheck.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        /// <param name="value">String to be checked.</param>
+        /// <param name="customMessage">Custom message to be put into the exception.</param>
+        /// <param name="messageParams">Parameters to be applied to the custom message.</param>
+        public static string nonEmpty(string value, string customMessage, params Object[] messageParams)
+        {
+            return nonEmpty(value, true, customMessage, messageParams);
+        }
+
+        /// <summary>
+        /// Nons the empty.
```
Comment (secondsun) [inline Core/AeroGear.Mobile/Core/Utils/SanityCheck.cs]: I think this summar can be more descriptive

**Pull request merged**